azurerm/internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -419,6 +420,10 @@ func resourceArmKeyVaultCertificateCreate(d *schema.ResourceData, meta interface
 			MinTimeout: 15 * time.Second,
 			Timeout:    d.Timeout(schema.TimeoutCreate),
 		}
+		if policy.IssuerParameters != nil && policy.IssuerParameters.Name != nil && *policy.IssuerParameters.Name != "Self" {
+			stateConf.PollInterval = 30 * time.Second
+			stateConf.NotFoundChecks = int(math.Round(float64(stateConf.Timeout) / float64(stateConf.PollInterval)))
+		}
```

Comment:
I'm not sure I understand the intent here?  Are we expecting certs to take rather longer with `IssuerParameters`?  Or do we expect `NotFound` to be received for a much greater time (i.e. more than the first 20 checks)?

Reply:
Quick answer: Our certificate issuer turned out to require their staff to validate the certificate creation, so the certificate can take a while to be available.

There is a "Certificate Operation" API where they give an ETA along the lines of

> Your order is being reviewed and certificate contents validated by DigiCert Validation staff, but should be issued shortly. For your reference, your DigiCert order id is [REDACTED]. Please check back around 2020-06-29T11:35Z.

There's also a "Request ID" and a "CA Name" (which is equal to the issuer).

The timestamp in the message text is not easy to parse, nor is it guaranteed that all issuers would include a timestamp in the message, so I think the best bet is to either allow longer timeouts or make them configurable.

Reply:
Whether an hour is enough really depends on the issuer. I haven't seen how any other resources or providers support custom timeout settings, but that might add the needed flexibility. Until then, I'll simply add a comment explaining what's going on and why.